daisy_workflows/image_build/install_package/cos/package_replacement/compile_debian_package.sh

```diff
@@ -0,0 +1,167 @@
+#!/bin/bash
+
+# This script compiles the debian packaging for google guest agent. Most of the
+# code here is sourced from: https://github.com/GoogleCloudPlatform/guest-test-infra/blob/master/packagebuild/daisy_startupscript_deb.sh
+# Compiling the debian package produces a set of binaries to be installed and
+# their respective installation paths.
+
+# Args: ./compile_debian_package [overlays_branch] [guest_agent_version]
+# Example: ./compile_debian_package master 20231214.00
+#     $1 [overlays_branch]: the COS milestone version (to apply the correct patches).
+#     $2 [guest_agent_version]: the guest agent version (to upgrade to).
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+apply_patches() {
+  # Download the repositories and apply COS specific patches.
+  echo -e "\nATTENTION: Downloading the board-overlays and guest-agent repos...\n"
+  git clone https://cos.googlesource.com/cos/overlays/board-overlays --branch ${overlays_branch}
+  git clone https://github.com/GoogleCloudPlatform/guest-agent.git --branch ${guest_agent_version}
+  mv ./board-overlays/project-lakitu/app-admin/google-guest-agent/files ./guest-agent
+  cd guest-agent
+
+  echo -e "\nATTENTION: Applying patches...\n"
+  search_dir=./files
+  for file in "$search_dir"/*
+  do
+    if [[ "$file" == *"patch"* ]]; then
+      git apply "$file"
+    fi
+  done
+}
+
+install_build_deps() {
+  # Install the build dependencies.
+  echo -e "\nATTENTION: Installing build deps...\n"
+  apt-get -y update
+  apt-get install -y --no-install-{suggests,recommends} git-core \
+    debhelper devscripts build-essential equivs libdistro-info-perl
+  export PKGNAME="google-guest-agent"
+  mk-build-deps -t "apt-get -o Debug::pkgProblemResolver=yes \
+    --no-install-recommends --yes" --install packaging/debian/control
+  dpkg-checkbuilddeps packaging/debian/control
+}
+
+find_debian_version() {
+  # Identify and store the debian version.
+  echo -e "\nATTENTION: Finding the debian version...\n"
+  cat /etc/debian_version
+  DEB_VERSION=$(</etc/debian_version)
+  # deb_version=${DEB}
+  if [[ "$DEB_VERSION" =~ "bookworm" ]]; then
+      export DEB="12"
+  else
+      export DEB="11"
+  fi
+}
+
+install_go(){
+  # Install the correct version of Go.
+  echo -e "\nATTENTION: Installing go...\n"
+  cd /workspace
+  source ./install_go.sh; install_go /tmp/go
+}
+
+upstream_tarball() {
+  # Create the build directory and the upstream tarball.
+  cd guest-agent
+  mkdir /tmp/debpackage
+  export BUILD_DIR="/tmp/debpackage"
+  export VERSION="112723"
```

Comment:
what's this version?

Reply:
it doesn't need to be an specific value, we need to communicate what they are.

Reply:
Added a comment to explain values